### PR TITLE
ISPN-14085 Allow to pass new list of indexed entities in update schema API

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-14085

Rework of #10447 
~~draft => not done~~ ready!

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 1]
```diff
@@ -811,7 +813,12 @@ private static void mutableAttributes(ConfigurationElement<?> element, Map<Strin
       prefix = prefix == null ? "" : element.elementName();
       for (Attribute<?> attribute : element.attributes().attributes()) {
          if (!attribute.isImmutable()) {
-            attributes.put(prefix + "." + attribute.getAttributeDefinition().name(), attribute);
+            AttributeDefinition<?> definition = attribute.getAttributeDefinition();
+            // even if mutable, we don't want expose this attribute to the user
+            if (AbstractTypedPropertiesConfiguration.PROPERTIES.equals(definition)) {
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:818]: I think the best thing to do is to completely remove `PROPERTIES` from the `IndexingConfiguration` and `QueryConfiguration` AttributeSets. leave the methods on the builders but make them log a warning and do nothing

Comment (fax4ever) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:818]: I would definitely like the idea of removing these properties. At the moment they are only used by our tests I think. Maybe we can do it in another issue, to have a better track of our changes. What do you think @tristantarrant?

**Comment by fax4ever** [comment]
Thank you for the review @tristantarrant!

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java (hunk 1 above)
Comment (fax4ever) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:818]: Created https://issues.redhat.com/browse/ISPN-14575 thanks

**Comment by fax4ever** [comment]
Rebased on current main to check the auto-merging of:
``` bash
Auto-merging server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java
Auto-merging core/src/main/java/org/infinispan/configuration/cache/Configuration.java
```

**Comment by fax4ever** [comment]
Thank you Tristan!

**Pull request merged**